packages/next-swc/crates/napi/src/lib.rs

```diff
@@ -73,19 +76,84 @@ shadow_rs::shadow!(build);
 static ALLOC: turbopack_binding::turbo::malloc::TurboMalloc =
     turbopack_binding::turbo::malloc::TurboMalloc;
 
+static LOG_FILE_MUTEX: Mutex<()> = Mutex::new(());
+static LOG_THROTTLE: Mutex<SystemTime> = Mutex::new(SystemTime::UNIX_EPOCH);
+static LOG_FILE_PATH: &str = ".next/turbopack.log";
+
 #[cfg(feature = "__internal_dhat-heap")]
 #[global_allocator]
 static ALLOC: dhat::Alloc = dhat::Alloc;
 
 #[cfg(not(target_arch = "wasm32"))]
 #[napi::module_init]
+
 fn init() {
-    if cfg!(debug_assertions) || env::var("SWC_DEBUG").unwrap_or_default() == "1" {
-        set_hook(Box::new(|panic_info| {
-            let backtrace = Backtrace::new();
-            println!("Panic: {:?}\nBacktrace: {:?}", panic_info, backtrace);
-        }));
-    }
+    use std::{fs::OpenOptions, io};
+
+    set_hook(Box::new(|panic_info| {
+        let mut last_error_time = LOG_THROTTLE.lock().unwrap();
+        if last_error_time.elapsed().unwrap().as_secs() < 1 {
+            // Throttle panic logging to once per second
+            return;
+        }
+        *last_error_time = SystemTime::now();
+
+        let backtrace = Backtrace::new();
+        let info = format!("Panic: {}\nBacktrace: {:?}", panic_info, backtrace);
+        if cfg!(debug_assertions) || env::var("SWC_DEBUG").unwrap_or_default() == "1" {
```

Comment:
Nit: avoid the `unwrap_or_default`

```suggestion
        if cfg!(debug_assertions) || env::var("SWC_DEBUG") == Ok("1") {
```